### PR TITLE
refactor(client): document and test create types utilities

### PR DESCRIPTION
**Pull request by victor-duarte**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

The chances for these utilities to have an incorrect output are low. Nevertheless adding some documentation and tests.
Related to #42256 since it seems a "cleanup" an update on a `ts` file. 

**Notes**
- `stateManagement.js`
  - Removed since logic was also defined in `create-types.ts`
  - Reference updates to existing `create-types.ts` seems to have not impacted scripts
- `createTypes.ts`
  - Removed default value for `ns` parameter since fallback to `annon` did not seem a real scenario and could lead to unwanted behaviors

### CODE REVIEW

File: client/src/utils/create-types.ts [hunk 1]
```diff
@@ -2,10 +2,18 @@ type CreateTypesType = {
...
-): CreateTypesType {
+/**
+ * Creates an object in which the `keys` represent the action names and the
+ * `values` the action type.
+ * {
+ *   action: actionType,
+ *   ...
+ * }
+ * @param {array} types Names of the actions.
+ * @param {string ns Name of the namespace.
```
Comment (ojeytonwilliams) [inline client/src/utils/create-types.ts:13]:
```suggestion
 * @param {string} ns Name of the namespace.
```
I'm not that familiar with JSDoc, but this looks like it should be closed.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
I only noticed one missing `}`.  Otherwise this looks great.

As a side question, what can we do with the JSDocs?  Sorry if this is a silly question, I was just curious since you'd added them.

**Review by ojeytonwilliams — APPROVED** [review]

**Comment by victor-duarte** [comment]
> I only noticed one missing `}`. Otherwise this looks great.
> 
> As a side question, what can we do with the JSDocs? Sorry if this is a silly question, I was just curious since you'd added them.

Hi @ojeytonwilliams, it complements the functions documentation with more specific description of the function input, output and what its doing. It is not easy to write, meaning sometimes they are not clear but the goal is to give devs an idea of what is going on without looking into the code.

Obviously these are not required since tests, TS and the notion that the code should be self descriptive should be a good set of docs.

Having said this, if `fcc` think that for consistency in the code base this should be removed just let me know if the usage of these should be not continued.

Also, depending on the editor you could see the following:

**With docs**

with-jsdocs

**Without docs**

without-jsdocs

**Comment by ojeytonwilliams** [comment]
Brilliant, thanks for explaining that.  I'm more than happy to have extra docs and more informative Intellisense.

We're never going to have a perfectly consistent codebase, and that's fine. Hopefully the presence of JSDocs for some functions will inspire people to write more.

**Review by huyenltnguyen — APPROVED** [review]
LGTM. Thanks for the refactoring and the documentation, Victor! 👍

**Pull request merged**